### PR TITLE
Rename settings constant value and related testing yml files for migration to Opensearch

### PR DESCRIPTION
**Pull request by andy840314**
##  opensearch-security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__
Rename `opendistro_security.` to `plugins.security` for settings.
Rename related yaml files for testing.

 4. __Why these changes are required?__ 

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java [hunk 1]
```diff
@@ -354,7 +354,7 @@ public void testLdapAuthorization() throws Exception {
                 .put(ConfigConstants.LDAP_AUTHZ_ROLEBASE, "ou=groups,o=TEST")
                 .put(ConfigConstants.LDAP_AUTHZ_ROLENAME, "cn")
                 .put(ConfigConstants.LDAP_AUTHZ_ROLESEARCH, "(uniqueMember={0})")
-                // .put("opendistro_security.authentication.authorization.ldap.userrolename",
```
Comment (cliu123) [inline src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java:357]: Can all the code commented out be removed?

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java [hunk 2]
```diff
@@ -1018,7 +1018,7 @@ public Settings additionalSettings() {
             return settingsFilter;
         }
 
-        settingsFilter.add("opendistro_security.*");
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:1021]: any downside of keeping `"opendistro_security.*"` in the filter?

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java (hunk 1 above)
Comment (andy840314) [reply src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java:357]: IMO, It should be.

File: src/main/java/org/opensearch/security/configuration/CompatConfig.java [hunk 3]
```diff
@@ -69,7 +69,7 @@ public boolean restAuthEnabled() {
             } else {
                 final boolean restDynamicallyDisabled = dcm.isRestAuthDisabled();
                 if (isTraceEnabled) {
-                    log.trace("opendistro_security.dynamic.disable_rest_auth {}", restDynamicallyDisabled);
+                    log.trace("plugins.security.dynamic.disable_rest_auth {}", restDynamicallyDisabled);
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/configuration/CompatConfig.java:72]: IMO, it refers to security config from the index, not settings

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java (hunk 2 above)
Comment (saratvemulapalli) [reply src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:1021]: +1 If there are any settings which are from opendistro they would be filtered out. 
I think we should keep it.

File: src/main/java/org/opensearch/security/configuration/CompatConfig.java (hunk 3 above)
Comment (andy840314) [reply src/main/java/org/opensearch/security/configuration/CompatConfig.java:72]: Correct, this should be reverted.

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java (hunk 2 above)
Comment (andy840314) [reply src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:1021]: Will add this line back.

File: src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java [hunk 4]
```diff
@@ -36,7 +36,7 @@ public SSLConfig(final boolean sslOnly, final boolean dualModeEnabled) {
         this.sslOnly = sslOnly;
         this.dualModeEnabled = dualModeEnabled;
         if (this.dualModeEnabled && !this.sslOnly) {
-            logger.warn("opendistro_security_config.ssl_dual_mode_enabled is enabled but opendistro_security.ssl_only mode is disabled. "
+            logger.warn("opendistro_security_config.ssl_dual_mode_enabled is enabled but plugins.security.ssl_only mode is disabled. "
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: Should refer to the actual settings that is disabled.

File: src/main/java/org/opensearch/security/support/ConfigConstants.java [hunk 5]
```diff
@@ -123,37 +123,37 @@
...
-    public static final String OPENDISTRO_SECURITY_AUDIT_THREADPOOL_SIZE = "opendistro_security.audit.threadpool.size";
-    public static final String OPENDISTRO_SECURITY_AUDIT_THREADPOOL_MAX_QUEUE_LEN = "opendistro_security.audit.threadpool.max_queue_len";
-    public static final String OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY = "opendistro_security.audit.log_request_body";
-    public static final String OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES = "opendistro_security.audit.resolve_indices";
-    public static final String OPENDISTRO_SECURITY_AUDIT_ENABLE_REST = "opendistro_security.audit.enable_rest";
-    public static final String OPENDISTRO_SECURITY_AUDIT_ENABLE_TRANSPORT = "opendistro_security.audit.enable_transport";
-    public static final String OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES = "opendistro_security.audit.config.disabled_transport_categories";
-    public static final String OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES = "opendistro_security.audit.config.disabled_rest_categories";
+    public static final String OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT = "plugins.security.audit.type";
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/support/ConfigConstants.java:135]: All deprecated settings should not be carried over.

Comment (andy840314) [reply src/main/java/org/opensearch/security/support/ConfigConstants.java:135]: For the deprecated settings, are all of them specified in `AuditConfig.java`? Are there any other deprecated settings?

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTestClientCert.java [hunk 6]
```diff
@@ -152,8 +152,8 @@ public void testLdapSslAuth() throws Exception {
                 .putList(ConfigConstants.LDAP_HOSTS, "localhost:636")
                 .put(ConfigConstants.LDAP_AUTHC_USERSEARCH, "(uid={0})")
                 .put(ConfigConstants.LDAPS_ENABLE_SSL, true)
-                .put("opendistro_security.ssl.transport.keystore_filepath", "/Users/temp/opendistro_security_integration_tests/ldap/ssl-root-ca/spock-keystore.jks")
-                .put("opendistro_security.ssl.transport.truststore_filepath", "/Users/temp/opendistro_security_integration_tests/ldap/ssl-root-ca/truststore.jks")
+                .put("plugins.security.ssl.transport.keystore_filepath", "/Users/temp/opendistro_security_integration_tests/ldap/ssl-root-ca/spock-keystore.jks")
```
Comment (vrozov) [inline src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTestClientCert.java:155]: use definition from `ConfigConstants`

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java (hunk 1 above)
Comment (andy840314) [reply src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java:357]: Any concerns of removing these?

File: src/test/resources/auditlog/endpoints/sink/configuration_no_default.yml [hunk 7]
```diff
@@ -1,32 +1,33 @@
...
-      endpoint8:
-        type: DeBug
-        config:
-          key: value
-      endpoint9:
-        type: external_opensearch
-        config:
-          endpoints: stringhere
\ No newline at end of file
+plugins:
```
Comment (vrozov) [inline src/test/resources/auditlog/endpoints/sink/configuration_no_default.yml:1]: can you do `plugins.security:` and keep everything else the same.

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTestClientCert.java (hunk 6 above)
Comment (andy840314) [reply src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTestClientCert.java:155]: There are 505 results for all of the testing files, It would take significant amount of time to replace all the the strings to `ConfigConstants`.

File: src/test/resources/legacy/securityconfig_v6/config.yml [hunk 8]
```diff
@@ -52,167 +52,168 @@
...
-            # Skip users matching a user name, a wildcard or a regex pattern
-            #skip_users: 
-            #  - 'cn=Michael Jackson,ou*people,o=TEST'
-            #  - '/\S*/'    
-      roles_from_another_ldap:
-        enabled: false
-        authorization_backend:
-          type: ldap #
-          #config goes here ...
+plugins:
```
Comment (vrozov) [inline src/test/resources/legacy/securityconfig_v6/config.yml:55]: Can it be moved to a separate PR? Also, this is legacy ES 6.x file, why it needs to be changed?

File: src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java (hunk 4 above)
Comment (andy840314) [reply src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: I don't get this. It is referring to the actual settings that is disabled.

File: src/test/resources/legacy/securityconfig_v6/config.yml (hunk 8 above)
Comment (andy840314) [reply src/test/resources/legacy/securityconfig_v6/config.yml:55]: Will be reverted.

File: src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java (hunk 4 above)
Comment (vrozov) [reply src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: Is that the setting that should be backward compatible?

Comment (hsiang9431-amzn) [reply src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: yes, `opendistro_security_config.ssl_dual_mode_enabled` is the only setting that needs to implement fallback value to be backward compatible

File: src/main/java/org/opensearch/security/support/ConfigConstants.java (hunk 5 above)
Comment (andy840314) [reply src/main/java/org/opensearch/security/support/ConfigConstants.java:135]: Done.

File: src/main/java/org/opensearch/security/configuration/CompatConfig.java (hunk 3 above)
Comment (andy840314) [reply src/main/java/org/opensearch/security/configuration/CompatConfig.java:72]: Done.

File: src/test/resources/auditlog/endpoints/sink/configuration_no_default.yml (hunk 7 above)
Comment (andy840314) [reply src/test/resources/auditlog/endpoints/sink/configuration_no_default.yml:1]: Done

File: src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java (hunk 1 above)
Comment (andy840314) [reply src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTest.java:357]: Let's not remove this since its not related to this PR

**Review by hsiang9431-amzn — APPROVED** [review]

File: src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java (hunk 4 above)
Comment (vrozov) [reply src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: So, if it is `opendistro_security.ssl_only` that is actually disabled, the warning message that refers to plugins.security.ssl_only is confusing. Can warning message refer to the actual setting that is disabled?

Comment (andy840314) [reply src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:39]: This warning specified `opendistro_security_config.ssl_dual_mode_enabled` is enabled and `opendistro_security.ssl_only mode is disabled`
as you can see in line 38 `if (this.dualModeEnabled && !this.sslOnly) {`
I don't think we should change this.

**Review by cliu123 — APPROVED** [review]
